cockatrice/src/spoilerbackgroundupdater.cpp

```diff
@@ -0,0 +1,241 @@
+#include <QDateTime>
+#include <QDebug>
+#include <QUrl>
+#include <QNetworkReply>
+#include <QMessageBox>
+#include <QFile>
+#include <QApplication>
+#include <QtConcurrent>
+#include "spoilerbackgroundupdater.h"
+#include "settingscache.h"
+#include "carddatabase.h"
+#include "main.h"
+
+SpoilerBackgroundUpdater::SpoilerBackgroundUpdater(QObject *apParent) : QObject(apParent)
+{
+    // User doesn't want to download spoilers... for now :)
+    mbIsActiveThread = settingsCache->getDownloadSpoilersStatus();
+    connect(settingsCache, SIGNAL(downloadSpoilerStatusChanged()), this, SLOT(changeActiveStatus()));
+
+    // If there is a change in the settings, update the timer so we know when to update correctly
+    connect(settingsCache, SIGNAL(downloadSpoilerTimeIndexChanged()), this, SLOT(handleNewTimeInterval()));
+
+    if (! mbIsActiveThread)
+    {
+        return;
+    }
+
+    // User wants spoilers, so lets see if we're in spoiler season or not.
+    downloadSpoilerSeasonStatusFileAndStartTimer();
+}
+
+SpoilerBackgroundUpdater::~SpoilerBackgroundUpdater()
+{
+    delete mpTimerForSpoilers;
```

Comment:
If the user disabled spoiler checking, mpTimerForSpoilers  will never be created, and trying to delete it will probably cause a crash.
Anyway, i saw that when you instanciate mpTimerForSpoilers and mpNetworkAccessManager you always pass "this" as the object parent, so you can just skip deleting these 2 objects since Qt will take care of deleting them when their parent gets deleted.
Sorry, i know i first asked you to explicitly delete mpNetworkAccessManager, i overlooked the code.